### PR TITLE
docs: clarify reloadable config option usage

### PR DESCRIPTION
**Pull request by jkirschner-hashicorp**
Improvements:
- Alter wording to make it more obvious when reading quickly that this is a list of reloadable options, not a list of non-reloadable option
- Mention that auto-reload is possible and how to configure it

### CODE REVIEW

File: website/content/docs/agent/config/index.mdx [hunk 1]
```diff
@@ -51,9 +51,12 @@ required ports and their default settings.
 
 ## Reloadable Configuration
 
-Reloading configuration does not reload all configuration items. The
-items which are reloaded include:
+The following agent configuration options are reloadable at runtime.
+To make a Consul agent automatically reload these options when changed in its configuration files on disk,
+start the agent with its [`auto_reload_config` configuration option](/consul/docs/agent/config/config-files#auto_reload_config) set to `true`.
+You can also manually trigger reload with the [reload command](/commands/reload).
```
Comment (trujillo-adam) [inline website/content/docs/agent/config/index.mdx:57]:
```suggestion
You can run the [`consul reload` command](/commands/reload) to manually reload configurations in the configuration directory.

To configure the agent to automatically reload configurations when files are updated on disk, set the [`auto_reload_config` configuration option](/consul/docs/agent/config/config-files#auto_reload_config) parameter to `true`. 

The following agent configuration options are reloadable at runtime: 
```
We should put "the following . . ." statements as close to the list of items as possible. I also took a stab at trimming the long sentence about the auto reload configuration.

File: website/content/docs/agent/config/index.mdx [hunk 2]
```diff
@@ -51,9 +51,12 @@ required ports and their default settings.
...
 
+Reloadable agent configuration options:
```
Comment (trujillo-adam) [inline website/content/docs/agent/config/index.mdx:59]:
```suggestion
```
This line becomes redundant with the previous suggestion.

**Review by trujillo-adam — COMMENTED** [review]
Suggested a few tweaks.

File: website/content/docs/agent/config/index.mdx (hunk 1 above)
Comment (jkirschner-hashicorp) [reply website/content/docs/agent/config/index.mdx:57]: I think it's important to lead by mentioning that only some options are reloadable to set expectations upfront.

Perhaps I can do something like this?
```suggestion
Some agent configuration options are reloadable at runtime.
You can run the [`consul reload` command](/commands/reload) to manually reload supported options from configuration files in the configuration directory.
To configure the agent to automatically reload configuration files updated on disk,
set the [`auto_reload_config` configuration option](/consul/docs/agent/config/config-files#auto_reload_config) parameter to `true`.

The following agent configuration options are reloadable at runtime: 
```

File: website/content/docs/agent/config/index.mdx [hunk 3]
```diff
@@ -51,9 +51,14 @@ required ports and their default settings.
...
-Reloading configuration does not reload all configuration items. The
-items which are reloaded include:
+Some agent configuration options are reloadable at runtime.
+You can run the [`consul reload` command](/commands/reload) to manually reload supported options from configuration files in the configuration directory.
+To configure the agent to automatically reload configuration files updated on disk,
+set the [`auto_reload_config` configuration option](/consul/docs/agent/config/config-files#auto_reload_config) parameter to `true`.
 
+The following agent configuration options are reloadable at runtime: 
+
+Reloadable agent configuration options:
```
Comment (trujillo-adam) [inline website/content/docs/agent/config/index.mdx:61]:
```suggestion
```
I think this is redundant.

**Review by trujillo-adam — APPROVED** [review]
One thing to address, but approving so you can merge as soon as you're ready.

File: website/content/docs/agent/config/index.mdx (hunk 3 above)
Comment (jkirschner-hashicorp) [reply website/content/docs/agent/config/index.mdx:61]: Oops. Yeah ... That got in there by accident!

Comment (jkirschner-hashicorp) [inline website/content/docs/agent/config/index.mdx:61]:
```suggestion
The following agent configuration options are reloadable at runtime: 
```

**Pull request merged**